tiledb/sm/array_schema/dimension.cc

```diff
@@ -1678,20 +1693,25 @@ Status Dimension::check_tile_extent() const {
   auto domain = (const T*)domain_.data();
   bool is_int = std::is_integral<T>::value;
 
-  // Check if tile extent is negative or 0
-  if (*tile_extent <= 0)
-    return LOG_STATUS(Status::DimensionError(
-        "Tile extent check failed; Tile extent must be greater than 0"));
-
   // Check if tile extent exceeds domain
   if (!is_int) {
+    // Check if tile extent is negative or 0
+    if (*tile_extent <= 0)
+      return LOG_STATUS(Status::DimensionError(
+          "Tile extent check failed; Tile extent must be greater than 0"));
+
     if (*tile_extent > (domain[1] - domain[0] + 1))
       return LOG_STATUS(
           Status::DimensionError("Tile extent check failed; Tile extent "
                                  "exceeds dimension domain range"));
   } else {
+    // Check if tile extent is negative or 0
```

Comment:
Typo in the comment (testing only against zero)?

Reply:
@stavrospapadopoulos Sounds good, I'll go in and add a few now that I see people are onboard with this change.